src/item_action.cpp

```diff
@@ -49,15 +51,19 @@ class actmenu_cb : public uimenu_callback {
             }
             // Don't write a message if unknown command was sent
             // Only when an inexistent tool was selected
-            auto itemless_action = item_actions.find( action );
-            if( itemless_action != item_actions.end() ) {
+            auto itemless_action = am.find( action );
+            if( itemless_action != am.end() ) {
                 popup( _("You do not have an item that can perform this action.") );
             }
             return false;
         }
 };
 
-item_action_map map_actions_to_items( player &p )
+item_action_generator::item_action_generator() = default;
+
+item_action_generator::~item_action_generator() = default;
```

Comment:
It's only the combination of `virtual` and `= default` that causes problems on older GCCs AFAIK. I haven't take a careful look at this PR, but what's broken that would indicate the defaulted special member functions?